thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/ThriftSerializationFormats.java

```diff
@@ -55,14 +62,43 @@
      */
     public static final SerializationFormat TEXT_NAMED_ENUM = SerializationFormat.of("ttext-named-enum");
 
-    private static final Set<SerializationFormat> THRIFT_FORMATS =
-            ImmutableSet.of(BINARY, COMPACT, JSON, TEXT, TEXT_NAMED_ENUM);
+    /**
+     * A way to lookup the related {@link TProtocolFactory} from a {@link SerializationFormat}.
+     * Entries are provided via registered SPI {@link ThriftProtocolFactoryProvider} implementations.
+     */
+    private static final Map<SerializationFormat, TProtocolFactory> knownProtocolFactories;
+
+    static {
+        final List<ThriftProtocolFactoryProvider> providers = ImmutableList.copyOf(
+                ServiceLoader.load(ThriftProtocolFactoryProvider.class,
+                                   ThriftProtocolFactoryProvider.class.getClassLoader()));
+        knownProtocolFactories = providers
+                .stream()
+                .map(ThriftProtocolFactoryProvider::entries)
+                .flatMap(Set::stream)
+                .collect(toImmutableMap(e -> e.serializationFormat, e -> e.tProtocolFactory));
+    }
+
+    /**
+     * Returns the {@link TProtocolFactory} for the specified {@link SerializationFormat}.
+     *
+     * @throws IllegalArgumentException if the specified {@link SerializationFormat} is not a
+     *         known Thrift serialization format
+     */
+    public static TProtocolFactory tProtocolFactory(SerializationFormat serializationFormat) {
```

Comment:
How about just `toProtocolFactory()` or `protocolFactory()`? That 't' looks somewhat awkward.